Fix fatal error in REST API response when an image has no attachment metadata.

mehulkaklotar opened the pull request:
## Summary
This bug is very unlikely to happen as it depends on attachment metadata. It is assumed that metadata should always be present for the attachment but when metadata is deleted from DB, we should ensure proper error handling by checking the data from REST.

Fixes #567 

## Relevant technical choices

Added more checks for `media_details` in response of REST API for attachments

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

mehulkaklotar committed:
#567 - check rest data has media_details as a key and is array before checking sizes

felixarntz reviewed (commented):
@mehulkaklotar Fix generally looks reasonable, but please add a unit test for the bug, so that we can verify it is fixed.

In the test, you can use the `rest_prepare_attachment` to e.g. make `$data['media_details']` an object. With that test, we should then verify whether the old code would fail vs the new code would pass.

mehulkaklotar committed:
#567 - add new unit test for checking media_details in attachment response when metadata is not present

mehulkaklotar commented:
@felixarntz I have added the new unit test for checking `media_details` in the attachment response data in the latest commit. Please review it once.

I have used the REST controller's `get_item` and deleted the attachment metadata before to get the `media_details` as an object because `rest_prepare_attachment` doesn't have control over changing the data in the response object in the test.

felixarntz reviewed (approved):
@mehulkaklotar Thank you, the test looks excellent!

adamsilverstein reviewed (approved):
Great!
